rootfs/etc/nginx/lua/sticky.lua

```diff
@@ -0,0 +1,119 @@
+local cipher = require("cipher")
+local util = require("util")
+local string = require("resty.string")
+local sha1_crypto = require("resty.sha1")
+local md5_crypto = require("resty.md5")
+
+local sticky_sessions = ngx.shared.sticky_sessions
+
+local DEFAULT_SESSION_COOKIE_NAME = "route"
+-- Currently STICKY_TIMEOUT never expires
+local STICKY_TIMEOUT = 0
+
+local _M = {}
+
+local sha1 = sha1_crypto:new()
+if not sha1 then
+  ngx.say("failed to create the sha1 object")
+  return
+end
+
+local md5 = md5_crypto:new()
+if not md5 then
+  ngx.say("failed to create the md5 object")
+  return
+end
+
+local function md5_digest(raw, eof)
+  md5:update(raw)
+  if eof then
+    return md5:final()
+  end
+  return nil
+end
+
+local function sha1_digest(raw, eof)
+  sha1:update(raw)
+  if eof then
+      return sha1:final()
+  end
+  return nil
+end
```

Comment:
It's just if you're encrypting long messages. Initially I was trying to make the modular so that it could be used in any cases. I can change this to simplify for now though.